### ISSUE TITLE
Add an option to select inflection regularity

### ISSUE THREAD

**Issue by psfinaki** [psfinaki/CheckYourCzech#181]
In Czech, comparatives are made from the positive adjectives with different degree of regularity ([1](http://vyuka.zsjarose.cz/getfile.php?FileID=3453), [2](https://cs.wikipedia.org/wiki/%C4%8Cesk%C3%A1_p%C5%99%C3%ADdavn%C3%A1_jm%C3%A9na#Stup%C5%88ov%C3%A1n%C3%AD), [3](https://www.pravopisne.cz/2016/10/pravidla-stupnovani-pridavnych-jmen/), [4](http://prirucka.ujc.cas.cz/?slovo=drz%C3%BD#nadpis2_1), [5](https://www.mojecestina.cz/article/2016092701-stupnovani-pridavnych-jmen)).

1. Most comparatives are easy to construct - they require changing _-ý_ or _-í_ with _-ejší_ or _-ější_ (e.g. _nový -> novější_ or _veselý -> veselejší_).
2. Some comparatives have different suffixes or lead to stem ending distortion, this is more or less regulated by related rules (e.g. _chytrý -> chytřejší_, _krátký -> kratší_,  _hezký -> hezčí_).
3. A small (looks like defined) group of adjectives are totally rebellious and irregular and undergo suppletion or severe stem change in comparative degree (e.g. _dobrý -> lepší_, _malý -> menší_).

So, currently the exercise offers just random adjectives for comparative construction - it would be useful to have an option of separately practicing regular and irregular adjectives in terms of comparative form.

One question is where to draw the line of regularity. I think, for the beginning the line can be drawn strictly under the first group of adjectives. This is because 3 group is very small anyway.

Therefore later the exercise can be extended in that sense - options like "require alternation", "require rearer suffix" or "totally irregular" can be added.

**Comment by psfinaki** [comment]
So currently the Scraper does this - when analyzing an adjective, it builds its "theoretical" comparative using some rules (`buildTheoreticalComparative` function) and then compares it to the real thing (`isRegular` function), later adding a boolean to the `IsRegular` column of the `adjectives` table.

The original plan was to leverage only one rule (about _-ejší_ and _ější suffixes), and along the way it turned out that it is easier to include a few more rules straight away rather than make a significant code change there later. So stem consonant alternation is also included.

On the client side, a simple select field is added, similar to Gender select field for nouns. `Regularity` field is added to the model. There were some issues though, looks like Fable has some troubles converting bools to strings and back - very sneaky problems that led to some hours of frustration and comments in the code.

Basically the whole code update is rather extension than modification which is good news. A lot of regularity-related tests were added.
